### DISCUSSION TITLE
may i use HttpProxyHandlerFactory as this?

### DISCUSSION THREAD

**Discussion by birthToSpring** [relayrides/pushy#963] [Q&A]
```java
final ApnsClient apnsClient = new ApnsClientBuilder()
    .setApnsServer(ApnsClientBuilder.DEVELOPMENT_APNS_HOST)
    .setSigningKey(ApnsSigningKey.loadFromPkcs8File(new File("/path/to/key.p8"),
            "TEAMID1234", "KEYID67890"))
    .setProxyHandlerFactory(new HttpProxyHandlerFactory(new InetSocketAddress("172.16.33.246", 8443)))
    .build();
```

172.16.33.246:8443 : this is my proxy ip and port !
But what i get is `{"reason":"BadCertificateEnvironment"}` and `{"msgId":"n2022060962a1baa4e4b09f8c3d201335","receiptTime":1654766245980,"receiptTimeStr":"2022-06-09 17:17:25","receiptStatus":2,"receiptDesc":"Stream closed before a reply was received"}`

**Comment by jchambers** [comment]
It looks like you're trying to use a signing key, but your proxy is presenting a client certificate to the APNs server. I think you need to configure your proxy to not present a client certificate.

**Reply by birthToSpring** [reply]
I use a NGINX and this module(https://github.com/chobits/ngx_http_proxy_connect_module) to build my proxy server, but i could not find you said the proxy client certificate

**Reply by jchambers** [reply]
I am, sadly, not in a position to provide technical support for third-party software (either nginx or that module).

As a debugging step, though, does your code work as expected if you do _not_ use a proxy? In other words, can we show that things _do_ work without a proxy, but then _do not_ work when the proxy is in use?

**Reply by birthToSpring** [reply]
Yes, my code is work well without use the proxy,  at the same time I also validated the proxy that it works well  to proxy other program code request. So i am very confused  where is go wrong!

**Reply by birthToSpring** [reply]
accurate to say that  if i remove the line code ".setProxyHandlerFactory(new HttpProxyHandlerFactory(new InetSocketAddress("172.16.33.246", 8443)))", my code will works well.
